ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsonleft/v1/tabsonleft/clientlibs/site/js/tabs.js

```diff
@@ -0,0 +1,482 @@
+                }
+                return result;
+            }
+        }
+
+        /**
+         * Handles tab keydown events
+         *
+         * @private
+         * @param {Object} event The keydown event
+         */
+        #onKeyDown(event) {
+            var index = this.#getTabIndexById(this.#_active);
+            var tabs = this.#getCachedTabs();
+            var lastIndex = tabs.length - 1;
+
+            switch (event.keyCode) {
+                case keyCodes.ARROW_LEFT:
+                case keyCodes.ARROW_UP:
+                    event.preventDefault();
+                    if (index > 0) {
+                        this.#navigateAndFocusTab(tabs[index - 1].id);
+                    }
+                    break;
+                case keyCodes.ARROW_RIGHT:
+                case keyCodes.ARROW_DOWN:
+                    event.preventDefault();
+                    if (index < lastIndex) {
+                        this.#navigateAndFocusTab(tabs[index + 1].id);
+                    }
+                    break;
+                case keyCodes.HOME:
+                    event.preventDefault();
+                    this.#navigateAndFocusTab(tabs[0].id);
+                    break;
+                case keyCodes.END:
+                    event.preventDefault();
+                    this.#navigateAndFocusTab(tabs[lastIndex].id);
+                    break;
+                default:
+                    return;
+            }
+        }
+
+        /**
+         * Refreshes the tab markup based on the current {@code Tabs#_active} index
+         *
+         * @private
+         */
+        #refreshActive() {
+            var tabpanels = this.#getCachedTabPanels();
+            var tabs = this.#getCachedTabs();
+            if (tabpanels) {
+                for (var i = 0; i < tabpanels.length; i++) {
+                    if (tabs[i].id === this.#_active) {
+                        tabpanels[i].classList.add(Tabs.selectors.active.tabpanel);
+                        tabpanels[i].removeAttribute(FormView.Constants.ARIA_HIDDEN);
+                        tabs[i].classList.add(Tabs.selectors.active.tab);
+                        tabs[i].setAttribute(FormView.Constants.ARIA_SELECTED, true);
+                        tabs[i].setAttribute(FormView.Constants.TABINDEX, "0");
+                    } else {
+                        tabpanels[i].classList.remove(Tabs.selectors.active.tabpanel);
+                        tabpanels[i].setAttribute(FormView.Constants.ARIA_HIDDEN, true);
+                        tabs[i].classList.remove(Tabs.selectors.active.tab);
+                        tabs[i].setAttribute(FormView.Constants.ARIA_SELECTED, false);
+                        tabs[i].setAttribute(FormView.Constants.TABINDEX, "-1");
+                    }
+                }
+            }
+        }
+
+        #getBeforeViewElement(instanceManager, instanceIndex) {
+            var result = {};
+            var instanceManagerId = instanceManager.getId();
```

Comment:
Please write testcases for repeatability in tabs on left , you can take example from tabs on top/wizard test cases to check which all cases needs to be checked.